File: C/1814-count-nice-pairs-in-an-array/main.go
package main

import "fmt"

func countNicePairs(nums []int) int {
	rev := make([]int, len(nums))
	for i := 0; i < len(nums); i++ {
		n := nums[i]
		b := 0
		for n > 0 {
			m := n % 10
			b = b*10 + m
			n /= 10
		}
		rev[i] = b
	}
	// fmt.Println(nums, rev)
	ret := 0
	m := map[int]int{}
	for i := 0; i < len(nums); i++ {
		m[nums[i]-rev[i]]++
	}
	// fmt.Println(m)
	for _, v := range m {
		ret += (v * (v - 1) / 2)
		ret %= 1000000007
	}

	return ret
}

func main() {
	fmt.Println(countNicePairs([]int{13, 10, 35, 24, 76}))
}

File: C/1814-count-nice-pairs-in-an-array/main.rs
struct Solution {}

use std::collections::HashMap;
impl Solution {
  pub fn count_nice_pairs(nums: Vec<i32>) -> i32 {
    let mut map: HashMap<i32, i32> = HashMap::new();
    let mut ret: i32 = 0;
    let mut ns = nums;
    ns.iter_mut().for_each(|n| {
      let m = *n;
      let mut rev: i32 = 0;
      while *n > 0 {
        rev = rev * 10 + *n % 10;
        *n /= 10;
      }
      let mut v = map.entry(rev - m).or_insert(0);
      ret += *v;
      ret %= 1000000007;
      *v += 1;
    });
    ret % 1000000007
  }
}
